### DISCUSSION TITLE
We need to testing metamask on stake, unstake, claim functions

### DISCUSSION THREAD

**Discussion by zksyncbusiness** [zkbusiness/zkstaking-platform#41] [Q&A] [answered]
@smart-developer-p 
Did u check all functions with metamask?

**Comment by zksyncbusiness** [comment]
@socialking0 
also, please check the functions as well with @smart-developer-p
please report to here when have done.

**Comment by CodeByStella** [comment]
Hi @zksyncstaking 
yes. I just checked all functions with metamask.

![image](https://github.com/user-attachments/assets/ce36b258-435c-43b3-a2a3-f1da64ade839)

![image](https://github.com/user-attachments/assets/7fa9817f-5729-4adb-b3a3-db8f1590549e)

![image](https://github.com/user-attachments/assets/6f233fb4-6501-4eee-9a5a-43a90f4dbf3f)

As you can see in those screenshots, all is working well on Metamask.

**Comment by blockchainSupport1125** [comment]
@zksyncstaking 
I checked with on my end too.
seem like all is good on metamask.
I think we need to check on other wallets.
:)

**Comment by CodeByStella** [comment, marked as answer]
so all is perfect on functions without any fail.

**Reply by zksyncbusiness** [reply]
> so all is perfect on functions without any fail.

Good
